### PR TITLE
add logic for creating/loading maps

### PR DESCRIPTION
**Pull request by jusrus01**
depends on https://github.com/jusrus01/TinyCiv/pull/11

### CODE REVIEW

File: src/TinyCiv.Server/Resources/Watery [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+----ww--------------
+---wwwwww-----------
+---www--------------
+-----w-------ww-----
+--------------------
+--------------------
+--wwww--------------
+--------------------
+------wwww----------
+--------------------
```
Comment (jusrus01) [inline src/TinyCiv.Server/Resources/Watery:20]: @Randwarf @devJUKI @jshimas 

map format

File: src/TinyCiv.Server/Services/MapLoader.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    private GameObjectType ResolveType(char c)
+    {
+        return c switch
+        {
+            '-' => GameObjectType.Empty,
+            'w' => GameObjectType.StaticWater,
+            'm' => GameObjectType.StaticMountain,
+            _ => throw new InvalidOperationException()
+        };
```
Comment (jusrus01) [inline src/TinyCiv.Server/Services/MapLoader.cs:68]: @Randwarf @devJUKI @jshimas 

valid objects in map

File: src/TinyCiv.Shared/Events/Client/StartGameClientEvent.cs [hunk 3]
```diff
@@ -1,6 +1,8 @@
+using TinyCiv.Shared.Game;
+
 namespace TinyCiv.Shared.Events.Client;
 
 /// <summary>
 /// Use to trigger game start
 /// </summary>
-public record StartGameClientEvent : ClientEvent;
\ No newline at end of file
+public record StartGameClientEvent(MapType MapType = MapType.Test) : ClientEvent;
```
Comment (jusrus01) [inline src/TinyCiv.Shared/Events/Client/StartGameClientEvent.cs:8]: @jshimas @Randwarf 

the one who starts the game can choose the map, "MapType.Test" - everything empty as was previously

**Comment by jusrus01** [comment]
no breaking changes, can be merged

**Review by devJUKI — APPROVED** [review]

**Pull request merged**